### PR TITLE
Random pistol option in pistolallocator

### PR DESCRIPTION
**Pull request by chriswoodle**
Would be nice to have a random pistol option for every round. My server is using the pistolallocator, so I added a pistol choice that selects a random pistol at each round. Tested it and seems to work correctly.

I'm new to modding so I hope this change is in the appropriate place.

### CODE REVIEW

**Comment by chriswoodle** [comment]
Try it out at pistol-retakes.kowop.net

File: scripting/retakes_pistolallocator.sp [hunk 1]
```diff
@@ -125,3 +131,7 @@ public int GunsMenuHandler(Menu menu, MenuAction action, int param1, int param2)
 public bool IsDefaultPistol(int weaponIndex) {
     return weaponIndex <= 2;
 }
+
+public bool IsRandomPistol(int weaponIndex) {
+    return weaponIndex == 7;
```
Comment (splewis) [inline scripting/retakes_pistolallocator.sp:136]: Replace 7 with `sizeof(g_PistolChoices) - 1`.

File: scripting/retakes_pistolallocator.sp [hunk 2]
```diff
@@ -80,6 +82,8 @@ public void PistolAllocator(ArrayList tPlayers, ArrayList ctPlayers, Bombsite bo
     for (int i = 0; i < tCount; i++) {
         int client = tPlayers.Get(i);
         int choice = g_PistolChoice[client];
+        if(IsRandomPistol(choice)) 
+            choice = GetRandomInt(0, 6);
```
Comment (splewis) [inline scripting/retakes_pistolallocator.sp:86]: Replace 6 with `sizeof(g_PistolChoices) - 2`. Same for the ct loop below.

File: scripting/retakes_pistolallocator.sp [hunk 3]
```diff
@@ -20,6 +20,8 @@ char g_PistolChoices[][][] = {
     { "weapon_tec9", "Tec-9" },
     { "weapon_fiveseven", "Five-Seven" },
     { "weapon_deagle", "Deagle" },
+    // Random Pistol
```
Comment (splewis) [inline scripting/retakes_pistolallocator.sp:23]: Can you adjust to comment to:

// Random pistol, must be last entry in the array.

Comment (chriswoodle) [reply scripting/retakes_pistolallocator.sp]: Changed

File: scripting/retakes_pistolallocator.sp (hunk 2 above)
Comment (chriswoodle) [reply scripting/retakes_pistolallocator.sp]: Changed

File: scripting/retakes_pistolallocator.sp (hunk 1 above)
Comment (chriswoodle) [reply scripting/retakes_pistolallocator.sp]: Changed

**Pull request merged**